pkg/apis/duck/v1/delivery_types.go

```diff
@@ -155,3 +155,52 @@ type DeliveryStatus struct {
 	// +optional
 	DeadLetterSinkURI *apis.URL `json:"deadLetterSinkUri,omitempty"`
 }
+
+func (d *DeliverySpec) Merge(other *DeliverySpec) *DeliverySpec {
+	if d == nil {
+		return other
+	}
+	if other == nil {
+		return d
+	}
+
+	r := &DeliverySpec{}
+
+	if d.Retry != nil {
+		r.Retry = d.Retry
+	} else {
+		r.Retry = other.Retry
+	}
+
+	if d.BackoffDelay != nil {
+		r.BackoffDelay = d.BackoffDelay
+	} else {
+		r.BackoffDelay = other.BackoffDelay
+	}
+
+	if d.BackoffPolicy != nil {
+		r.BackoffPolicy = d.BackoffPolicy
+	} else {
+		r.BackoffPolicy = other.BackoffPolicy
+	}
+
+	if d.DeadLetterSink != nil {
+		r.DeadLetterSink = d.DeadLetterSink
+	} else {
+		r.DeadLetterSink = other.DeadLetterSink
+	}
```

Comment:
I like the idea of merging the different values.

Can we add some doc, to clarify the behavior, that the _own_ value has preference over the `_other_` value?

Question: would this change the spec - the merge? 
Regardless, it does make sense to merge